### DISCUSSION TITLE
No logs coming from the hello-display pod in the drift detection example

### DISCUSSION THREAD

**Discussion by bgalvao** [kubeflow/kfserving#2934] [Q&A]
Help appreciated! I've deployed the KNative CRDs and was running the [example notebook for drift detection](https://github.com/kserve/kserve/blob/master/docs/samples/drift-detection/alibi-detect/cifar10/cifar10_drift.ipynb). However, I'm not getting any logs from the `tfserving-cifar10-predictor` pod nor from the `hello-display` pod.

The predict function of the notebook that has a POST request to the InferenceService does not raise any HTTP error either 😖 

So I am confused as to why this is happening without an error.

Version info

- kserve at v0.10.1
- knative serving at v1.7.0
- kserve eventing v1.8.8

**Comment by elukey** [comment]
@bgalvao Hi! There are a lot of improvements on the logging front in 0.11, but it is still in rc candidate release. If you have time to test it and report back it would be great!

**Reply by bgalvao** [reply]
hmm, I could test it in a local kind cluster! Where can I find the changelog regarding logging?

**Reply by elukey** [reply]
https://github.com/kserve/kserve/releases/tag/v0.11.0-rc0

**Reply by bgalvao** [reply]
@elukey

TLDR, tried it, it's throwing me an error

Ok, I did the following steps

1. created kind cluster with `kindest/node:v1.26.3`.
2. downloaded and ran the [`quick_install.sh` script from `v0.11.0-rc0`](https://raw.githubusercontent.com/kserve/kserve/v0.11.0-rc0/hack/quick_install.sh)
3. Installed KNative Eventing v1.10.0 - CRDs and core.
4. Optional: installed [MetalLB v0.13.7](https://raw.githubusercontent.com/metallb/metallb/v0.13.7/config/manifests/metallb-native.yaml) for a local load balancer and [configured address pools according to the guide from kind](https://kind.sigs.k8s.io/docs/user/loadbalancer/#setup-address-pool-used-by-loadbalancers).

I tried both with the MetalLB enabled and disabled (in this case I used a `kubectl port-forward`), and I get the same error traceback running the [CIFAR10 drift detection demo notebook](https://github.com/kserve/kserve/blob/v0.11.0-rc0/docs/samples/drift-detection/alibi-detect/cifar10/cifar10_drift.ipynb). Redacted traceback

```
ConnectionRefusedError                    Traceback (most recent call last)
NewConnectionError                        Traceback (most recent call last)
ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=80): Max retries exceeded with url: /v1/models/tfserving-cifar10:predict (Caused by NewConnectionError(': Failed to establish a new connection: [Errno 111] Connection refused'))
```

Mind you, getting the cluster IP while having MetalLB enabled worked just fine

```python
CLUSTER_IPS=!(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
CLUSTER_IP=CLUSTER_IPS[0]
print(CLUSTER_IP)
```
